cfpb/consumerfinance.gov
# Implement nicer buttons with icons

Implements buttons with icons designs from https://cfpb.github.io/design-system/components/buttons#buttons-with-icons-1.

Tries to fix https://github.com/cfpb/design-system/issues/1070.

## How to test this PR

Run a local server and visit http://localhost:8000/admin/template_debug/v1/call_to_action/ to debug the call to action module.

||
|-|
|![image](https://user-images.githubusercontent.com/654645/132055622-f931c245-ab38-4a6c-8265-93ccce3b36c6.png)|

A good live page to test with is http://localhost:8000/about-us/blog/buying-home-first-step-check-your-credit/; this includes both the call to action buttons as well as some regular links that get decorated.

## Screenshots

|Normal|Hover|Tabbed|
|-|-|-|
|![image](https://user-images.githubusercontent.com/654645/132055660-5142d9b1-8b4a-4248-ab81-54af9dc3a955.png)|![image](https://user-images.githubusercontent.com/654645/132055685-e4c68a55-8172-456f-b9fb-48e2131e5771.png)|![image](https://user-images.githubusercontent.com/654645/132055713-c8299aea-3b8d-4c60-92e5-1d299932f9fd.png)|
|![image](https://user-images.githubusercontent.com/654645/132055775-4bf5b705-7fb2-4a54-9786-9b98755e3e98.png)|![image](https://user-images.githubusercontent.com/654645/132055759-3e8fadfb-9f64-4614-93b0-9a9cc965ddd8.png)|![image](https://user-images.githubusercontent.com/654645/132055738-695b0d1b-002e-46a3-a2d0-796b6f857060.png)|

## Notes and todos

~~@anselmbradford I think there's some remaining work here.~~ ~~The hover state seems off if you hover over just the outer edge and not the middle, but I'm not sure the intended design since the buttons in the DS don't have that extra padding.~~

~~Are there examples of regular buttons (not call to action) on the site somewhere to test with?~~

~~This PR also needs some tests before it can be merged.~~

## Checklist
